keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/credential/Ssha256PasswordHashProvider.java

```diff
@@ -0,0 +1,87 @@
+package io.cloudtrust.keycloak.credential;
+
+import org.keycloak.common.util.Base64;
+import org.keycloak.credential.hash.PasswordHashProvider;
+import org.keycloak.models.PasswordPolicy;
+import org.keycloak.models.credential.PasswordCredentialModel;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+/**
+ * Implementation of the Salted SHA256 password hash algorithm.
+ */
+public class Ssha256PasswordHashProvider implements PasswordHashProvider {
+    private final String algorithm = "ssha256";
+
+    @Override
+    public boolean policyCheck(PasswordPolicy policy, PasswordCredentialModel credential) {
+        return policy != null && credential.getPasswordCredentialData().getAlgorithm().equals(algorithm);
+    }
+
+    @Override
+    public PasswordCredentialModel encodedCredential(String rawPassword, int iterations) {
+        byte[] salt = new byte[32];
+        SecureRandom random = new SecureRandom();
+        random.nextBytes(salt);
+
+        byte[] toEncode;
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            byte[] hash = digest.digest(concatBytes(rawPassword.getBytes(), salt));
+
+            toEncode = concatBytes(hash, salt);
+            String value = "{SSHA256}" + Base64.encodeBytes(toEncode);
+
+            return PasswordCredentialModel.createFromValues(algorithm, salt, 1, value);
+        } catch (IOException | NoSuchAlgorithmException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public boolean verify(String rawPassword, PasswordCredentialModel credential) {
+        if (!credential.getPasswordSecretData().getValue().startsWith("{SSHA256}")) {
```

Comment:
Things cannot be null here ?

Reply:
To be honest I don't really know, but based on Keycloak's implementation of [PBKDF2](https://github.com/keycloak/keycloak/blob/8f2c3a744713db18033f1b4f2f004eeb70b1fb08/server-spi-private/src/main/java/org/keycloak/credential/hash/Pbkdf2PasswordHashProvider.java#97) and [Argon2](https://github.com/keycloak/keycloak/blob/8f2c3a744713db18033f1b4f2f004eeb70b1fb08/crypto/default/src/main/java/org/keycloak/crypto/hash/Argon2PasswordHashProvider.java#88), I'd say this case is not covered.